src/lib.rs

```diff
@@ -1,7 +1,28 @@
 #![forbid(unsafe_code)]
+#![no_std]
 
-use std::error;
-use std::fmt;
+#[cfg(all(not(feature = "alloc"), not(feature = "std")))]
+compile_error!("feature \"std\" and feature \"alloc\" mut be valid at least one");
```

Comment:
This seems unnecessary. We should just default to the `alloc` stuff if the `std` feature is not enabled. There's no point in having two features when one will do.

Reply:
You are right. I will remove 'alloc' feature.